File: backend/routes/chat_routes.py
from flask import Blueprint, request, jsonify
from services.chat_service import ask_ai, rate_answer, flag_answer, get_chat_history

chat_bp = Blueprint('chat', __name__)

@chat_bp.post("/ask")
def ask():
    data = request.get_json()

    user_id = data.get("user_id")
    question = data.get("question")

    if not user_id or not question:
        return jsonify({"error": "user_id and question are required."}), 400
    
    response, status = ask_ai(user_id, question)
    return jsonify(response), status


@chat_bp.post("/rate")
def rate():
    data = request.get_json()

    answer_id = data.get("answer_id")
    rating = data.get("rating")

    if answer_id is None or rating is None:
        return jsonify({"error": "answer_id and rating are required."}), 400
    
    response, status = rate_answer(answer_id, int(rating))
    return jsonify(response), status

@chat_bp.post("/flag")
def flag():
    data = request.get_json()

    answer_id = data.get("answer_id")
    reason = data.get("reason")

    if answer_id is None:
        return jsonify({"error": "answer_id is required."}), 400
    
    response, status = flag_answer(answer_id, reason)
    return jsonify(response), status

@chat_bp.get("/history/<user_id>")
def history(user_id):
    response, status = get_chat_history(user_id)
    return jsonify(response), status

File: backend/routes/chat_routes.js
const express = require('express');
const router = express.Router();
const { askAi, rateAnswer, reviewKbAnswer, flagAnswer, getChatHistory } = require('../services/chat_service');

router.post("/ask", async (req, res) => {
  const { user_id, question } = req.body;
  
  if (!user_id || !question) {
    return res.status(400).json({ error: "user_id and question are required." });
  }
  
  const [response, status] = await askAi(user_id, question);
  return res.status(status).json(response);
});

router.post("/rate", async (req, res) => {
  const { answer_id, rating } = req.body;
  
  if (answer_id === undefined || rating === undefined) {
    return res.status(400).json({ error: "answer_id and rating are required." });
  }
  
  const [response, status] = await rateAnswer(answer_id, parseInt(rating));
  return res.status(status).json(response);
});

router.post("/review-kb", async (req, res) => {
  const { answer_id, rating } = req.body;
  
  if (answer_id === undefined || rating === undefined) {
    return res.status(400).json({ error: "answer_id and rating are required." });
  }
  
  const [response, status] = await reviewKbAnswer(answer_id, parseInt(rating));
  return res.status(status).json(response);
});

router.post("/flag", async (req, res) => {
  const { answer_id, reason } = req.body;
  
  if (answer_id === undefined) {
    return res.status(400).json({ error: "answer_id is required." });
  }
  
  const [response, status] = await flagAnswer(answer_id, reason);
  return res.status(status).json(response);
});

router.get("/history/:user_id", async (req, res) => {
  const [response, status] = await getChatHistory(req.params.user_id);
  return res.status(status).json(response);
});

module.exports = router;
